### PR TITLE
feat: updated the date time to support timezone

### PR DESCRIPTION
**Pull request by Chandru-HM**
This PR is for ticket https://github.com/awslabs/iot-app-kit/issues/2663 and added the timezone support for the date and time in xy-plot, KPI, and resource explorer

## Verifying Changes
changes are not changing anything, date-time values are now adjusted to the timezone.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/e2e/tests/trendCursors/trendCursors.spec.ts [hunk 1]
```diff
@@ -185,8 +185,18 @@ test('can drag a trend cursor', async ({ page }) => {
   expect(initialTrendCursorDateString).not.toEqual(
     updatedTrendCursorDateString
   );
-  const initialDate = new Date(initialTrendCursorDateString);
-  const updatedDate = new Date(updatedTrendCursorDateString);
+  const initialDate = new Date(
+    initialTrendCursorDateString
+      .split('\n')
+      .join('T')
+      .replace(/(\d+)\/(\d+)\/(\d+)/, '$3-$2-$1')
```
Comment (jmbuss) [inline packages/dashboard/e2e/tests/trendCursors/trendCursors.spec.ts:192]: does the date-fns library not support this?

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/modeledDataStreamTable/createModeledDataStreamColumnDefinitions.tsx [hunk 2]
```diff
@@ -33,12 +34,17 @@ export function createModeledDataStreamColumnDefinitions(
...
       header: 'Latest value time',
       cell: ({ latestValueTime }) => {
+        if (!latestValueTime) return '-';
         if (latestValueTime && isNumeric(latestValueTime)) {
-          return getFormattedDateTimeFromEpoch(
-            Number(round(latestValueTime, significantDigits))
+          return (
+            <DateTime
+              dateTime={
+                Number(round(latestValueTime, significantDigits)) * 1000
```
Comment (jmbuss) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/modeledDataStreamTable/createModeledDataStreamColumnDefinitions.tsx:42]: we're rounding the date?

File: packages/react-components/src/components/chart/legend/table/columnDefinitions/trendCursor/trendCursorHeader.tsx [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
 export const TrendCursorColumnHeader = ({
   date,
   color,
 }: TrendCursorColumnHeaderOptions) => {
+  const dateTime = useDateTime(date, FULL_DATE_TIME.replace(',', ''));
   return (
     <div className='base-chart-legend-tc-header-container'>
       <div>
-        <span>{date.toLocaleDateString()}</span>
+        <span>{dateTime.split(' ')[0]}</span>
```
Comment (jmbuss) [inline packages/react-components/src/components/chart/legend/table/columnDefinitions/trendCursor/trendCursorHeader.tsx:18]: Ca we use the date-fns formatter here?

File: packages/dashboard/e2e/tests/trendCursors/trendCursors.spec.ts (hunk 1 above)
Comment (Chandru-HM) [reply packages/dashboard/e2e/tests/trendCursors/trendCursors.spec.ts]: supports. updated.

File: packages/react-components/src/components/chart/legend/table/columnDefinitions/trendCursor/trendCursorHeader.tsx (hunk 3 above)
Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/table/columnDefinitions/trendCursor/trendCursorHeader.tsx]: yes, Done

**Pull request closed without merging**